flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBase.java

```diff
@@ -517,16 +519,13 @@ public void initializeState(FunctionInitializationContext context) throws Except
 					LOG.debug("Using the following offsets: {}", restoredState);
 				}
 			}
-			if (restoredState != null && restoredState.isEmpty()) {
-				restoredState = null;
-			}
 		} else {
 			LOG.info("No restore state for FlinkKafkaConsumer.");
 		}
 	}
 
 	@Override
-	public void snapshotState(FunctionSnapshotContext context) throws Exception {
+	public final void snapshotState(FunctionSnapshotContext context) throws Exception {
```

Comment:
@stevenzwu the snapshotState method was actually never intended to be overriden, hence making it final here to state that clearly. For example, the verrsion-specific implementations for `FlinkKafkaConsumerBase` may override that and have incorrect implementations, where as our tests would never realize it.